pkg/healthcheck/healthcheck.go

```diff
@@ -962,24 +962,41 @@ func validateDataPlanePodReporting(pods []*pb.Pod) error {
 }
 
 func checkUnschedulablePods(pods []corev1.Pod) error {
+	var podName []string
 	for _, pod := range pods {
 		for _, node := range pod.Status.Conditions {
-			if node.Status == "False" && node.Message == "no nodes available to schedule pods" {
-				return fmt.Errorf("No node available for the pod %s", pod.Name)
+			if strings.Contains(node.Message, "Insufficient cpu") || (node.Status == "False" && node.Message == "no nodes available to schedule pods") {
```

Comment:
There are other possible failures, like insufficient memory. I believe it's better to just check for `Reason == "FailedSheduling"` instead of checking for the error message strings, so we can cover any possible failure.

Reply:
i am not  getting reason as Reason == "FailedSheduling" instead i am getting node.Reason == "ContainersNotReady", should i check for it

Reply:
Could you please clarify how can I reproduce that? Also it would be helpful to get the output of `kubectl -n linkerd get events` right after the install fails.

Reply:
i was able to check for the reason in the two cases it comes out to be the same i.e.  node.Reason == "Unschedulable"

Reply:
> Also it would be helpful to get the output of `kubectl -n linkerd get events` right after the install fails.

as if the output of `kubectl -n linkerd get events` after the check ?

Reply:
Reason `Unschedulable` is actually fine. One last minor change: could you rename the `node` var in that loop to something like `condition`?

Reply:
something  like `isNodeAvailable`  ?

Reply:
or `nodeAvailability`

Reply:
sorry i misread